src/Filter.ts

```diff
@@ -0,0 +1,110 @@
+export interface FilterArgs {
+	[ name: string ]: Filter | RegExp | string | (() => boolean);
+}
+
+export default class Filter {
+	[index: string]: any;
+	type: string;
+	args: any[];
+
+	constructor();
+	constructor(filterArg: string);
+	constructor(filterArg: () => any);
+	constructor(filterArg: FilterArgs);
+	constructor(filterArg?: any) {
+		if (typeof filterArg === 'string' || typeof filterArg === 'function') {
+			this.type = typeof filterArg;
+			this.args = [filterArg];
+		}
+		else if (filterArg) {
+			var filter = this;
```

Comment:
The style guide says to use `const` for immutable variable and to use `let` for mutable variables https://github.com/dojo/guidelines/blob/master/STYLE.md#variables